applications/main/gpio/scenes/gpio_scene_usb_uart.c

```diff
@@ -39,10 +45,18 @@ void gpio_scene_usb_uart_on_enter(void* context) {
 bool gpio_scene_usb_uart_on_event(void* context, SceneManagerEvent event) {
     GpioApp* app = context;
     if(event.type == SceneManagerEventTypeCustom) {
-        scene_manager_set_scene_state(app->scene_manager, GpioSceneUsbUart, 1);
-        scene_manager_next_scene(app->scene_manager, GpioSceneUsbUartCfg);
+        if(event.event == GpioUsbUartEventConfig) {
```

Comment:
Now it looks overcomplicated. Looks like you want to combine exit and uart scene